File: cpp/04-control-flow/conditionals.cpp
#include <iostream>
#include <string>

int main() {
    int score = 85;

    // if / else if / else chain
    if (score >= 90) {
        std::cout << "Grade: A" << std::endl;
    } else if (score >= 80) {
        std::cout << "Grade: B" << std::endl;
    } else if (score >= 70) {
        std::cout << "Grade: C" << std::endl;
    } else {
        std::cout << "Grade: F" << std::endl;
    }

    // Ternary conditional operator: condition ? value_if_true : value_if_false
    int number = 7;
    std::string parity = (number % 2 == 0) ? "even" : "odd";
    std::cout << number << " is " << parity << std::endl;

    // Logical operators combine conditions
    int age = 25;
    bool hasLicense = true;
    if (age >= 18 && hasLicense) {
        std::cout << "Allowed to drive" << std::endl;
    }

    // if with initializer (C++17): remainder only exists inside this if
    if (int remainder = score % 10; remainder == 5) {
        std::cout << "Score ends in 5 (remainder " << remainder << ")" << std::endl;
    }

    return 0;
}

File: typescript/04-control-flow/conditionals.ts
function classify(score: number): string {
    if (score >= 90) {
        return "A";
    } else if (score >= 80) {
        return "B";
    } else if (score >= 70) {
        return "C";
    } else {
        return "F";
    }
}

const scores: number[] = [95, 82, 71, 64];

for (const score of scores) {
    console.log(`Score ${score} => Grade ${classify(score)}`);
}
